robotbase/src/main/java/frc/robot/RobotContainer.java

```diff
@@ -76,12 +83,19 @@ public RobotContainer() {
    * passing it to a {@link edu.wpi.first.wpilibj2.command.button.JoystickButton}.
    */
   private void configureButtonBindings() {
-    // m_gunnerControls.m_shootButton.whenPressed(command)
-    m_driverControls.m_invertButton.whenPressed(new InvertDriveCommand(m_visionSub, m_driverControls));
-    m_driverControls.m_changePipelineButton.whileHeld(new VisionChangePipelineCommand(m_visionSub));
-    // m_gunnerControls.m_rightTrigger.whileActiveContinuous(new ShootCommand(m_shootSub, m_gunnerControls));
-    m_gunnerControls.m_leftTrigger.whileActiveContinuous(new IntakePickupBallCommand(m_intakeSub, m_gunnerControls));
-    m_gunnerControls.m_yButton.whenPressed(new IntakeToggleDirectionCommand(m_intakeSub));
+    Map<String, ClosedLoopSubsystem> subsystems = new HashMap<>();
```

Comment:
I don't like the idea of stringly typed code when it's not necessary. You should just make a class that has a member variable for each subsystem and initialize them to null. Then you don't have to cast when you use them, either. But again, you're jumping through a lot of hoops just to keep two functions out of RobotContainer.

Reply:
I started moving the subsystems into a separate class, but I realized it was probably a bit better to pass in an array. While getting the contents of configureButtonBindings is one of the reasons for making this change, it also is helpful for testing, as it makes it so that you can comment out one line to remove all commands dependent on a certain subsystem in one line. Once we bind all of the commands to their respective controls, the method would get huge, so I think that getting in out of RobotContainer is a near necessary thing.